service/topology_coordinator.cc

```diff
@@ -1014,6 +1014,12 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
 
             switch (trinfo.stage) {
                 case locator::tablet_transition_stage::allow_write_both_read_old:
+                    if (action_failed(tablet_state.barriers[trinfo.stage])) {
+                        if (check_excluded_replicas()) {
+                            transition_to_with_barrier(locator::tablet_transition_stage::revert_migration);
```

Comment:
I think we could jump to revert_migration without barrier, and revert_migration which is not preceded by cleanup could also avoid do_barrier(). This should make reverting more robust since it would only require group0 quorum. But this is optimization.